pkg/cluster/meta/topology.go

```diff
@@ -889,9 +887,78 @@ func (topo *TopologySpecification) dirConflictsDetect() error {
 	return nil
 }
 
+// CountDir counts for dir paths used by any instance in the cluster with the same
+// prefix, useful to find potential path conflicts
+func (topo *ClusterSpecification) CountDir(targetHost, dirPrefix string) int {
+	dirTypes := []string{
+		"DataDir",
+		"DeployDir",
+		"LogDir",
+	}
+
+	// path -> count
+	dirStats := make(map[string]int)
+	count := 0
+	topoSpec := reflect.ValueOf(topo).Elem()
+
+	for i := 0; i < topoSpec.NumField(); i++ {
+		if isSkipField(topoSpec.Field(i)) {
+			continue
+		}
+
+		compSpecs := topoSpec.Field(i)
+		for index := 0; index < compSpecs.Len(); index++ {
+			compSpec := compSpecs.Index(index)
+			// Directory conflicts
+			for _, dirType := range dirTypes {
+				if j, found := findField(compSpec, dirType); found {
+					dir := compSpec.Field(j).String()
+					host := compSpec.FieldByName("Host").String()
+
+					if dir == "" {
+						continue
+					}
+					if !strings.HasPrefix(dir, "/") {
+						switch dirType {
+						case "DataDir":
+							if !strings.HasPrefix(dir, topo.GlobalOptions.DataDir+"/") {
+								dir = filepath.Join(topo.GlobalOptions.DataDir, dir)
```

Comment:
Our data directory is relative to `DeployDir`.

Reply:
This only works if the data_dir in global options is an absolute one, the data_dir in instance scope itself is already filled with custom default values at unmarshaling